src/modules/claim/resolvers/credential.resolver.ts

```diff
@@ -45,13 +47,42 @@ export class RoleCredentialResolver implements CredentialResolver {
       subject,
       claimType,
     });
-    if (
-      !claim ||
-      !claim.registrationTypes.includes(RegistrationTypes.OffChain)
-    ) {
+    if (!claim || !this.isOffchainClaim(claim)) {
       return null;
     }
 
+    return this.claimToJwt(claim);
+  }
+
+  async eip191JwtsOf(subject: string): Promise<RoleEIP191JWT[]> {
+    const claims = await this.claimService.getBySubject({ subject });
+    return claims
+      .filter(Boolean)
+      .filter(this.isOffchainClaim)
+      .map(this.claimToJwt);
+  }
+
+  async credentialsOf(
+    subject: string
+  ): Promise<VerifiableCredential<RoleCredentialSubject>[]> {
+    const claims = await this.claimService.getBySubject({
+      subject,
+    });
+
+    return claims
+      .map(
+        (claim) =>
+          claim?.vp
+            ?.verifiableCredential?.[0] as VerifiableCredential<RoleCredentialSubject>
+      )
+      .filter(Boolean);
+  }
```

Comment:
Why are these two methods added? They don't seem to be used.

Reply:
they are part of CredentialResolver interface from ew-credentials

Reply:
Do you think it's better to get the credentials from the `claimService` or from the DID document / IPFS in this case?

Reply:
Perhaps yes, because it allow to verify issuer which prefer to not publish his credential

Reply:
We decided that issuers must publish their credentials. This is to not rely on a potentially centralized ssi-hub.

Therefore, I think this doesn't align then with our implementation from passport-did-auth and I think we should have alignment so that the behaviour is consistent across the stack. https://github.com/energywebfoundation/passport-did-auth/blob/2aa3e85ab316f474186c8eaf81f878c4c351155e/lib/RoleCredentialResolver.ts#L142

Reply:
@jrhender I think this is completely fine. Agreed that the credentials needs to be published so that it is verified through the IPFS resolution approach, but then there could be a case where user wants to fetch the cached credentials and interact with sis-hub by choice.

Reply:
IMO we can use these methods for simply fetching the credentials based on the request by holder ( for example displaying the holder's credential in switchboard ) and not use this for any verification.

Reply:
@jrhender If credentials should read from document can we use IpfsCredentialResolver from vc-verification for this? 
And also should issuer and revoker resolvers in ssi-hub also read from chain?